index.d.ts

```diff
@@ -0,0 +1,490 @@
+export function gte_<A>(x: undefined): (y: undefined) => boolean;
+export function gte_<A>(x: boolean, y: boolean): boolean;
+export function gte_<A>(x: boolean): (y: boolean) => boolean;
+export function gte_<A>(x: number, y: number): boolean;
+export function gte_<A>(x: number): (y: number) => boolean;
+export function gte_<A>(x: Date, y: Date): boolean;
+export function gte_<A>(x: Date): (y: Date) => boolean;
+export function gte_<A>(x: string, y: string): boolean;
+export function gte_<A>(x: string): (y: string) => boolean;
+export function gte_<A>(x: Array<A>, y: Array<A>): boolean;
+export function gte_<A>(x: Array<A>): (y: Array<A>) => boolean;
+export function gte_<A>(x: IArguments, y: IArguments): boolean;
+export function gte_<A>(x: IArguments): (y: IArguments) => boolean;
+export function gte_<A>(x: Ord<A>, y: Ord<A>): boolean;
+export function gte_<A>(x: Ord<A>): (y: Ord<A>) => boolean;
+
+export function min<A>(x: Date, y: Date): Date;
+export function min<A>(x: Date): (y: Date) => Date;
+export function min<A>(x: number, y: number): number;
+export function min<A>(x: number): (y: number) => number;
+export function min<A>(x: string, y: string): string;
+export function min<A>(x: string): (y: string) => string;
+export function min<A>(x: Ord<A>, y: Ord<A>): Ord<A>;
+export function min<A>(x: Ord<A>): (y: Ord<A>) => Ord<A>;
+
+export function max<A>(x: Date, y: Date): Date;
+export function max<A>(x: Date): (y: Date) => Date;
+export function max<A>(x: number, y: number): number;
+export function max<A>(x: number): (y: number) => number;
+export function max<A>(x: string, y: string): string;
+export function max<A>(x: string): (y: string) => string;
+export function max<A>(x: Ord<A>, y: Ord<A>): Ord<A>;
+export function max<A>(x: Ord<A>): (y: Ord<A>) => Ord<A>;
+
+export function id<A>(typeRep: TypeRep): (A) => A;
+export function id<A>(typeRep: TypeRep): Category<A>;
+
+export function concat<A>(x: string, y: string): string;
+export function concat<A>(x: string): (y: string) => string;
+export function concat<A>(x: Array<A>, y: Array<A>): Array<A>;
+export function concat<A>(x: Array<A>): (y: Array<A>) => Array<A>;
+export function concat<A>(x: Semigroup<A>, y: Semigroup<A>): Semigroup<A>;
+export function concat<A>(x: Semigroup<A>): (y: Semigroup<A>) => Semigroup<A>;
+
+export function empty<A>(typeRep: TypeRep): Monoid<A>;
+
+export function map<A, B>(f: (A) => B, xs: Array<A>): Array<B>;
+export function map<A, B>(f: (A) => B): (xs: Array<A>) => Array<B>;
+export function map<A, B>(f: (A) => B, functor: Functor<A>): Functor<B>;
+export function map<A, B>(f: (A) => B): (functor: Functor<A>) => Functor<B>;
+
+export function bimap<A, B, C, D>(f: (A) => B, g: (C) => D, bifunctor: Bifunctor<A, C>): Bifunctor<B, D>;
+export function bimap<A, B, C, D>(f: (A) => B, g: (C) => D): (bifunctor: Bifunctor<A, C>) => Bifunctor<B, D>;
+export function bimap<A, B, C, D>(f: (A) => B): AwaitingTwo<(C) => D, Bifunctor<A, C>, Bifunctor<B, D>>;
```

Comment:
OK, I know what you mean. The current branch haven't updated the `(A) => B` to `(a: A) => B`, so that `A` is considered a variable name and its type is `any`.